ci/drone/drone.yml

```diff
@@ -5,7 +5,17 @@ name: test-aarch64-linux-musl
 platform:
   arch: arm64
 
+# Fetch full Git history and tags so `git describe` will work.
+# Documentation: https://docs.drone.io/pipeline/docker/syntax/cloning/
+clone:
+  depth: 0
+
 steps:
+- name: fetch-tags
+  image: alpine/git
+  commands:
+  - git fetch --tags
```

Comment:
any reason this can't just go into the script? Also couldn't the other scripts be solved the same way? keeping the logic in the script rather than the CI configuration feels more maintainable for me, easier to migrate between them

Reply:
This was copied straight from the [Drone documentation on the topic of `--tags`](https://docs.drone.io/pipeline/docker/syntax/cloning/#the---tags-flag). I think it also makes sense to do it from the Drone script - fewer lines too - so I will make that change. Missing tags only occur in Drone, so I don't think it should go in other non-Drone scripts.

As for the clone depth, that is controlled by the CI configs.

Reply:
Right but presumably you could use a git command to fetch more depth if needed, right? I'd rather make the script support being run from a shallow clone than fiddle with N different CI configs.

Reply:
Okay, I will retest with the following added to all CI scripts and undo config changes:
```
git fetch --unshallow || true  # Fails if repository is already complete
git fetch --tags
```

Reply:
That seems to work.